Geometry/GEMGeometryBuilder/src/GEMGeometryBuilderFromDDD.cc

```diff
@@ -356,26 +354,24 @@ void GEMGeometryBuilderFromDDD::build(GEMGeometry& theGeometry,
   std::vector<GEMSuperChamber*> superChambers;
 
   while (doChambers) {
-    MuonBaseNumber mbn = muonConstants.geoHistoryToBaseNumber(fv.history());
-    cms::GEMNumberingScheme gemnum(muonConstants.values());
-    gemnum.baseNumberToUnitNumber(mbn);
-    GEMDetId detIdCh = GEMDetId(gemnum.getDetId());
+    MuonDDDNumbering mdddnum(muonConstants);
+    GEMNumberingScheme gemNum(muonConstants);
+    int rawidCh = gemNum.baseNumberToUnitNumber(mdddnum.geoHistoryToBaseNumber(fv.history()));
+    GEMDetId detIdCh = GEMDetId(rawidCh);
 
     if (detIdCh.layer() == 1) {  // only make superChambers when doing layer 1
       GEMSuperChamber* gemSuperChamber = buildSuperChamber(fv, detIdCh);
       superChambers.push_back(gemSuperChamber);
     }
 
-    GEMChamber* gemChamber = buildChamber(fv, detIdCh);
+    GEMChamber* gemChamber = ((detIdCh.station() == GEMDetId::minStationId0) ? nullptr : buildChamber(fv, detIdCh));
```

Comment:
If `gemChamber` is set to `nullptr`, it looks like the program could crash at line 377. Looking throughout this file, there seem to be several places where `gemChamber` is used even though it could be null. I don't follow all the logic, and it may be the dangerous cases are somehow avoided, but some of the use of `gemChamber` looks dangerous.

Reply:
This part of the code need to be modified for GE0. Ian Watson will take care of that. The station number will not be 0 unless it is GE0. One possibility is that I make it the same as before and it will be needed when GE0 code will come in.

Reply:
I shall comment this part and restore the original line and ask Ian later on to make that part work as he did for the ddd part.